src/puppetlabs/puppetdb/mq_listener.clj

```diff
@@ -359,29 +284,38 @@
     (.start)))
 
 (defn send-delayed-message [{:keys [^ConnectionFactory conn-pool endpoint]}]
-  (fn [message-to-convert message-properties]
-    (with-open [connection (create-connection conn-pool)
-                session (create-session connection)]
-      (let [q (create-queue session endpoint)]
-        (with-open [producer (.createProducer session q)]
-
-          (.send producer (mq/to-jms-message session message-to-convert message-properties)))))))
+  (fn [message exception]
+    (time!
+     (global-metric :retry-persistence-time)
+     (with-open [connection (create-connection conn-pool)
+                 session (create-session connection)]
+       (let [q (create-queue session endpoint)
+             message-with-error (-> message
+                                    (annotate-with-attempt exception)
+                                    json/generate-string)]
+         (with-open [producer (.createProducer session q)]
+           (->> (mq/delay-property 1 :seconds)
```

Comment:
I started with 60 seconds here, which seems right if we have some number of immediate retries. Not sure how exactly that would look, maybe 4 quick retries, then delay it for a minute, then 4 quick retries etc until we get 16? This value can't be 60 seconds right now without breaking tests. Specifically our full stack tests. They hit a constraint violation and need to retry. Our code that blocks waiting for results doesn't wait for 60+ seconds.